docs: add CLAUDE.md with project guidance for Claude Code (#91)
pokujs/c8: CLAUDE.md

# @pokujs/c8

@pokujs/c8 is a [Poku](https://github.com/wellwelwel/poku) plugin for V8 code coverage using [c8](https://github.com/bcoe/c8).

## Commands

```sh
npm ci          # install dependencies
npm run build   # compile TypeScript → lib/
npm test        # build then run e2e tests via poku
npm run lint    # check formatting with prettier
npm run lint:fix  # auto-fix formatting
```

To run a single test file:

```sh
npx tsx <path-to-test-file>
```

## Architecture

**Source files (`src/`):**

- `index.ts` — exports the `coverage()` factory. `setup()` creates a temp dir, sets `NODE_V8_COVERAGE`, and merges config. `teardown()` calls c8's `Report` and optionally `checkCoverages`, then cleans up the temp dir.
- `config.ts` — `loadConfig()` auto-discovers `.c8rc`/`.nycrc` config files (JSON, JSONC, TOML, YAML). Kebab-case keys from config files are mapped to camelCase `CoverageOptions` fields.
- `types.ts` — `CoverageOptions` type defining all plugin options.

**Build output:** TypeScript compiles to `lib/` (Node16 module format, declarations included). Only `lib/` is published.

**Tests (`test/`):**

> - All tests are e2e, run via `npm test`. Each test uses `inspectPoku()` to spawn a full poku process against a fixture project in `test/__fixtures__/`.

Reviewer: ```suggestion
- Run tests via `npm test`. Each test uses `inspectPoku()` to spawn a full poku process against a fixture project in `test/__fixtures__/`.
```

> - All tests are e2e, run via `npm test`. Each test uses `inspectPoku()` to spawn a full poku process against a fixture project in `test/__fixtures__/e2e/`.

Reviewer: This is not a rule, there is no need to require that tests always be end-to-end (The same idea applies to my previous comment about specifying paths as `<...>/e2e`) 😅

Author: I believe you are not understanding the purpose of the file. This file is not for dictating rules, but rather for providing context about the CURRENT status of the application so that the harness and LLM can more easily find files and work on the project.
If it were a rule, it would be in .claude/rules and not in CLAUDE.md. 🥲

Author: As the project progresses and other types of tests are implemented, this file will also change.

# CLAUDE.md

> This file provides guidance to Claude Code (claude.ai/code) when working with code in this repository.

Reviewer: ```suggestion
# @pokujs/c8

@pokujs/c8 is a [Poku](https://github.com/wellwelwel/poku) plugin for V8 code coverage using [c8](https://github.com/bcoe/c8).
```

## Commands

```sh

> npm ci          # install dependencies
> npm run build   # compile TypeScript → lib/

Reviewer: ```suggestion
```

- `npm test` already builds the project.

Author: Even though the `test` command already does everything, it's still important to give context to the LLM if it only wants to validate a part of the build.

npm test        # build then run e2e tests via poku
npm run lint    # check formatting with prettier
npm run lint:fix  # auto-fix formatting
```

To run a single test file:

> ```sh
> node_modules/poku/lib/bin/index.js test/e2e/coverage-basic.test.ts
> ```

Reviewer: ```suggestion
```sh
npx tsx <path-to-test-file>
```
```

## Architecture

> This is `@pokujs/c8`, a [Poku](https://poku.io) plugin that wraps [c8](https://github.com/bcoe/c8) to provide V8 code coverage. It implements Poku's `PokuPlugin` interface with two lifecycle hooks.

Reviewer: ```suggestion
```

**Source files (`src/`):**

- `index.ts` — exports the `coverage()` factory. `setup()` creates a temp dir, sets `NODE_V8_COVERAGE`, and merges config. `teardown()` calls c8's `Report` and optionally `checkCoverages`, then cleans up the temp dir.
- `config.ts` — `loadConfig()` auto-discovers `.c8rc`/`.nycrc` config files (JSON, JSONC, TOML, YAML). Kebab-case keys from config files are mapped to camelCase `CoverageOptions` fields.
- `types.ts` — `CoverageOptions` type defining all plugin options.

**Build output:** TypeScript compiles to `lib/` (Node16 module format, declarations included). Only `lib/` is published.

**Tests (`test/`):**

> - All tests are e2e, run via `poku --sequential`. Each test uses `inspectPoku()` to spawn a full poku process against a fixture project in `test/__fixtures__/e2e/`.

Reviewer: ```suggestion
- All tests run sequentially via `npm test`. Each test uses `inspectPoku()` to spawn a full poku process against a fixture project in `test/__fixtures__/`.
```

Author: The correct way to format it here is `test/__fixtures__/e2e/`. I'm just going to replace it to make use of `npm test`

Reviewer: > The correct way to format it here is `test/__fixtures__/e2e/`. I'm just going to replace it to make use of `npm test`

It's not a rule that 100% of the tests must be E2E. If some scenario is more viable as a simple test or even integration, regression, etc., the ideal is to keep the idea of "where to find" instead of replicating the file and directory tree 🙋🏻‍♂️

Author: I agree, but in this case it's only indicating what's happening in the current file tree, which only contains the e2e folder. This only serves to indicate where LLM should look for tests, not where it should write new ones. 👍

> - Fixtures live in `test/__fixtures__/e2e/` with their own configs under `configs/` and source under `src/`.

Reviewer: ```suggestion
- Fixtures live in `test/__fixtures__/` with their own configs under `configs/` and source under `src/`.
```

Author: This approach, relying solely on `test/__fixtures__/`, does not reflect the reality of the project.

Reviewer: > This approach, relying solely on `test/__fixtures__/`, does not reflect the reality of the project.

This is what I mean by "teaching where to find" instead of replicating the file tree of directories and files in a markdown file. If at some point another directory comes up, the duplication or renaming of files and directories is fragile and becomes hard to maintain, besides in the end both bringing more effort for LLMs and also spending more tokens than a simple `ls`.

Author: I disagree. As I mentioned in some comments, this file serves to tell LLM where to find files. Passing a directory that doesn't match the project's reality won't help, it will actually hinder it. As the project progresses, this file also changes, resolving the point you raised.

**Config priority (highest to lowest):** `--coverageConfig` CLI flag → `config` plugin option → auto-discovery → c8 defaults. Plugin options always override file config values.

> **Monocart support** is opt-in via `experimental: ['monocart']` and requires the `monocart-coverage-reports` peer dependency.

Reviewer: ```suggestion
**Monocart support** is a c8 experimental feature and requires the `monocart-coverage-reports` optional peer dependency.
```
